debugger.jpda.projectsui/src/org/netbeans/modules/debugger/jpda/projectsui/BreakpointAnnotationProvider.java

```diff
@@ -412,6 +412,10 @@ private void addAnnotationTo(JPDABreakpoint b, FileObject fo, int[] lines) {
         }
         List<DebuggerBreakpointAnnotation> annotations = new ArrayList<DebuggerBreakpointAnnotation>();
         for (int l : lines) {
+            if (l < 1){
```

Comment:
what if `l` does not correspond to any line, i.e. pas document's end ?

Reply:
I wansn't having issues with that, so I didn't add that condition. Do you want me to add it?

Reply:
There appears to be not definitive way to determine if the index is past the document's end without getting the list from `lc.getLineSet().getLines()` (which may be expensive?). We may be able to do it if we added a method to `Line.Set` to get the count, but I don't know if that would be practical since that might involve creating every line in the `Line.Set`. Do you have any suggestions?

Reply:
No... `IIOOB` catch is there, no need to add perf penalty for searching the line set. Thanks.